webui/react/src/stores/userSettings.tsx

```diff
@@ -232,6 +233,11 @@ export class UserSettingsStore extends PollingStore {
       const response = await getUserSetting({ signal: this.canceler?.signal });
       this.updateSettingsFromResponse(response);
     } catch (error) {
+      // if the error is due to an expired token, don't hide public message, to
+      // ensure a redirect to the SSO providers
+      if (error instanceof DetError && isRemoteUserTokenExpired(error)) {
+        handleError(error);
```

Comment:
i think we wanna avoid running the bottom `handleError` if if block `handleError` is called right? if so, we can use `eles block` for the bottom one